core/include/sched_round_robin.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    Sched_RR
+ * @ingroup     sys
+ * @brief       This module module provides round robin scheduling within
+ *              for all runnable threads within each not masked priority.
+ *              Priority 0 is masked by default.
+ *              There is no fainess involved in this scheuler. It schedules,
+ *              when the timer ticks. Keep it as simple as possible.
+ *
+ * @note        start using start_schedule_rr()
+ *
+ * @{
+ *
+ * @file
+ * @brief       Round Robin Scheduler
+ *
+ * @author      Karl Fessel <karl.fessel@ovgu.de>
+ *
+ */
+
+#if !defined(SCHED_RR_TIMER) || defined(DOXYGEN)
+/**
+ * @brief Time between round robin calls in µs
+ *
+ * @details Defaults to 1ms
+ */
+#define SCHED_RR_TIMER 1000
+#endif
+
+#if !defined(SCHED_RR_MASK) || defined(DOXYGEN)
+/**
+ * @brief masks off priorities that should not be scheduled default: 0 is masked
+ */
+#define SCHED_RR_MASK 1<<0
+#endif
+
+/**
+ * @brief remove the round robin timer
+ */
+void remove_sched_round_robin(void);
+
+/**
+ * @brief setup the round robin timer for prio
+ */
+void set_sched_round_robin(uint8_t prio);
+
+
+/**
+ * @brief setup round robin for priority if possible an sensible
+ */
+void check_set_sched_round_robin(uint8_t prio);
```

Comment:
From the documentation it is not clear what the difference is to `set_sched_round_robin()`. Do we really need two functions?

IMO we should try not to provide largely overlapping functions to the public. It will confuse users (they have to decide on a specific version of a function-family) and it increases the maintenance burden.

(Note: I'm certainly a huge fan of many small internal functions that help de-duplicating and structuring code. My comment above only applies for the public APIs.)

Reply:
you are right i already thought about making `set_sched_round_robin()` static an suggest inlineing to the compiler